components/brave_shields/browser/ad_block_resource_provider.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_AD_BLOCK_RESOURCE_PROVIDER_H_
+#define BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_AD_BLOCK_RESOURCE_PROVIDER_H_
+
+#include <string>
+
+#include "base/callback.h"
+#include "base/observer_list.h"
+#include "brave/components/brave_component_updater/browser/dat_file_util.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+using brave_component_updater::DATFileDataBuffer;
+
+namespace brave_shields {
+
+class AdBlockResourceProvider {
```

Comment:
What's the difference between `AdBlockResourceProvider` and `AdBlockSourceProvider`?
Can you please add comments for all important classes/interfaces/methods?

Reply:
yeah I see, just want some explanations in the code, because at some point I've got lost in Sources/Resources :D

Reply:
@iefremov I think we're going to change Sourceprovider -> FiltersProvider, but some comments would also be helpful

Reply:
I'm renaming `AdBlockSourceProvider` to `AdBlockFiltersProvider` to reduce confusion here as well